aws/src/main/python/runPilates/lambda_function.py

```diff
@@ -0,0 +1,318 @@
+
+
+def terminate_instance(instance_ids):
+    return ec2.terminate_instances(InstanceIds=instance_ids)
+
+
+def lambda_handler(event, context):
+    titled = event.get('title', 'hostname-test')
+    if titled is None:
+        return "Unable to start the run, runName is required. Please restart with appropriate runName."
+
+    initial_urbansim_input = event.get('initial_urbansim_input', None)
+    if initial_urbansim_input is None:
+        return "Unable to start the run, initialS3UrbansimInput is required. Please restart with appropriate arguments."
+
+    initial_urbansim_output = event.get('initial_urbansim_output', None)
+
+    branch = event.get('branch', BRANCH_DEFAULT)
+    commit_id = event.get('commit', COMMIT_DEFAULT)
+
+    start_year = event.get('start_year', START_YEAR_DEFAULT)
+    count_of_years = event.get('count_of_years', COUNT_OF_YEARS_DEFAULT)
+    beam_iteration_length = event.get('beam_it_len', BEAM_IT_LEN_DEFAULT)
+    urbansim_iteration_length = event.get('urbansim_it_len', URBANSIM_IT_LEN_DEFAULT)
+    pilates_image_version = event.get('pilates_image_version', PILATES_IMAGE_VERSION_DEFAULT)
+    pilates_image_name = event.get('pilates_image_name', PILATES_IMAGE_NAME_DEFAULT)
+    pilates_scenario_name = event.get('pilates_scenario_name', PILATES_SCENARIO_NAME_DEFAULT)
+    initial_skims_path = event.get('initial_skims_path', '')
+    in_year_output = event.get('in_year_output', IN_YEAR_OUTPUT_DEFAULT)
+    output_bucket_base_path = event.get('output_bucket_base_path', OUTPUT_BUCKET_BASE_PATH_DEFAULT)
+
+    configs = event.get('configs', CONFIG_DEFAULT)
+    batch = event.get('batch', TRUE)
+    max_ram = event.get('max_ram', MAXRAM_DEFAULT)
+    instance_type = event.get('instance_type', os.environ['INSTANCE_TYPE'])
+    volume_size = event.get('storage_size', 64)
+    shutdown_wait = event.get('shutdown_wait', SHUTDOWN_DEFAULT)
+    region = event.get('region', os.environ['REGION'])
+    shutdown_behaviour = event.get('shutdown_behaviour', os.environ['SHUTDOWN_BEHAVIOUR'])
+    sigopt_client_id = event.get('sigopt_client_id', os.environ['SIGOPT_CLIENT_ID'])
+    sigopt_dev_id = event.get('sigopt_dev_id', os.environ['SIGOPT_DEV_ID'])
+    end_script = event.get('end_script', END_SCRIPT_DEFAULT)
+
+    if initial_urbansim_output is None and initial_skims_path is '':
+        return "Unable to start run, initialS3UrbansimOutput (initial beam data) or initialSkimsPath (initial skims file) should be specified."
+
+    if instance_type not in instance_types:
+        return "Unable to start run, {instance_type} instance type not supported.".format(instance_type=instance_type)
+
+    if shutdown_behaviour not in shutdown_behaviours:
+        shutdown_behaviour = os.environ['SHUTDOWN_BEHAVIOUR']
+
+    if volume_size < 64 or volume_size > 256:
+        volume_size = 64
+
+    selected_script = RUN_PILATES_SCRIPT
+
+    params = configs
+    if batch == TRUE:
+        params = [params.replace(',', ' ')]
```

Comment:
yes, this was really strange things to read.
now it is gone for good because we do not need batch runs in pilates.
but it is still in beam deploy lambda function